backend-go/config.go

```diff
@@ -1,232 +1,104 @@
-		ErrorUnset:       false,
-		WeaklyTypedInput: true,
-		Result:           &config,
-	}
-
-	decoder, err := mapstructure.NewDecoder(decoderConfig)
-	if err != nil {
-		// Decoder for TelestionBaseConfig inference could not be initialized!
-		return nil, err
-	}
-
-	if err := decoder.Decode(*mapping); err != nil {
-		// TelestionBaseConfig could not be inferred from given map!
-		return nil, err
-	}
-
-	return &config, nil
-}
-
-// Loads and parses the service [Config] from different configuration sources in the following order:
-//
-// 1. `overwriteArgs`
-// 2. command line arguments
-// 3. environment variables
-// 4. default configuration, if `--dev` is passed in the steps from above
-// 5. configuration file, if `CONFIG_FILE` parameter is defined, readable and parsable
-func assembleConfig(overwriteArgs map[string]string) (*Config, error) {
-	config := &map[string]any{}
-
-	// add config params from passed service options
-	updateWith(config, &overwriteArgs)
-	// add config params from command line arguments
-	updateWith(config, cliConfig())
-	// add config params from environment variables
-	updateWith(config, envConfig())
-
-	// add default config if "dev" configuration is defined
-	if dev, ok := (*config)["DEV"].(bool); ok && dev {
-		fmt.Println("Running in development mode. Using default values for missing environment variables.")
-		dc, err := devModeDefaultConfig()
-		if err != nil {
-			return nil, err
+// MergeMaps merges multiple maps into one.
+// If a key is present in multiple maps, the value of the last map is used.
+func MergeMaps(maps ...map[string]any) map[string]any {
+	res := make(map[string]any)
+	for _, m := range maps {
+		for k, v := range m {
+			res[k] = v
 		}
-		updateWith(config, dc)
 	}
-
-	// add config file parameters if "CONFIG_FILE" is defined and readable
-	if configPath, ok := (*config)["CONFIG_FILE"].(string); ok && len(configPath) != 0 {
-		fc, err := fileConfig(configPath)
-		if err != nil {
-			return nil, err
-		}
-		updateWith(config, fc)
-	}
-
-	// verify if configuration is valid
-	if err := assertContainsMinimalConfig(*config); err != nil {
-		return nil, err
-	}
-
-	return parseConfig(config)
+	return res
 }
 
-// Adds entries from updates to base that don't exist in base.
-func updateWith[V any | string](base *map[string]any, updates *map[string]V) {
-	for k, v := range *updates {
-		if _, contained := (*base)[k]; !contained {
-			(*base)[k] = v
+// CliArgsMap returns a map of all CLI arguments in the form of --KEY=VALUE or --KEY VALUE
+// where KEY is the uppercase key and VALUE is the value.
+// If the argument is a flag (i.e., does not have a value), the value is true.
+// Positional arguments are ignored.
+func CliArgsMap() map[string]any {
+	res := make(map[string]any)
+	for /* don't include executable ($0) => start at 1 */ i := 1; i < len(os.Args); i++ {
+		key := os.Args[i]
+
+		if !strings.HasPrefix(key, "--") {
+			// ignore positional arguments
+			continue
 		}
```

Comment:
I think solution 1 is a fair suggestion. I would be strongly opposed to the second suggestion, as this would (a) create a lot more complexity (what if the positional argument comes before any key-value ones?), be non-standard (which can also come with vulnerabilities as standard CLI escape / sanitization libraries could lead to injections of wrong values), etc.

Reply:
Done